python/cudf_polars/cudf_polars/experimental/expressions.py

```diff
@@ -0,0 +1,439 @@
+    """
+    expr: Expr
+    exprs: list[Expr]
+    if agg.name == "count":
+        # Chunkwise stage
+        columns, input_ir, partition_info = select(
+            [agg],
+            input_ir,
+            partition_info,
+            names=names,
+            repartition=True,
+        )
+
+        # Combined stage
+        (column,) = columns
+        columns, input_ir, partition_info = select(
+            [Agg(agg.dtype, "sum", None, column)],
+            input_ir,
+            partition_info,
+            names=names,
+        )
+        (expr,) = columns
+    elif agg.name == "mean":
+        # Chunkwise stage
+        exprs = [
+            Agg(agg.dtype, "sum", None, *agg.children),
+            Agg(agg.dtype, "count", None, *agg.children),
+        ]
+        columns, input_ir, partition_info = select(
+            exprs,
+            input_ir,
+            partition_info,
+            names=names,
+            repartition=True,
+        )
+
+        # Combined stage
+        exprs = [
+            BinOp(
+                agg.dtype,
+                plc.binaryop.BinaryOperator.DIV,
+                *(Agg(agg.dtype, "sum", None, column) for column in columns),
+            )
+        ]
+        columns, input_ir, partition_info = select(
+            exprs,
+            input_ir,
+            partition_info,
+            names=names,
+            repartition=True,
+        )
+        (expr,) = columns
+    elif agg.name == "n_unique":
+        # Get uniques and shuffle (if necessary)
+        # TODO: Should this be a tree reduction by default?
+        (child,) = agg.children
+        pi = partition_info[input_ir]
+        if pi.count > 1 and [ne.value for ne in pi.partitioned_on] != [input_ir]:
+            from cudf_polars.experimental.shuffle import Shuffle
+
+            children, input_ir, partition_info = select(
+                [UnaryFunction(agg.dtype, "unique", (False,), child)],
+                input_ir,
+                partition_info,
+                names=names,
+            )
+            (child,) = children
+            agg = agg.reconstruct([child])
+            shuffle_on = (NamedExpr(next(names), child),)
+            input_ir = Shuffle(
+                input_ir.schema,
+                shuffle_on,
+                config_options,
+                input_ir,
+            )
+            partition_info[input_ir] = PartitionInfo(
+                count=pi.count,
+                partitioned_on=shuffle_on,
+            )
+
+        # Chunkwise stage
+        columns, input_ir, partition_info = select(
+            [Cast(agg.dtype, agg)],
+            input_ir,
+            partition_info,
+            names=names,
+            repartition=True,
+        )
+
+        # Combined stage
+        (column,) = columns
+        columns, input_ir, partition_info = select(
+            [Agg(agg.dtype, "sum", None, column)],
+            input_ir,
+            partition_info,
+            names=names,
+        )
```

Comment:
To understand this, is there a reason we can't put the cast into the final sum aggregation?